### PR TITLE
Adds move/rename feature

### PR DESCRIPTION
**Pull request by appukuttan-shailesh**
Implements #2 
New features:
- Move a file
- Rename a file

Could potentially be extended (if needed) to allow moving data between containers.

### CODE REVIEW

File: hbp_archive.py [hunk 1]
```diff
@@ -137,6 +137,31 @@ def read(self, decode='utf-8', accept=[]):
...
+            Target directory where the file is to be moved.
+        new_name : string, optional
+            New name to be assigned to file (including extension, if any)
+        overwrite : boolean, optional
+            Specify if any already existing file needs to be overwritten.
+        """
+        if self.container:
+            self.container.move(self.name, target_directory=target_directory, new_name=new_name, overwrite=overwrite)
+        else:
+            raise Exception("Parent container not known, unable to download")
```
Comment (apdavison) [inline hbp_archive.py]: "move" not "download"

File: hbp_archive.py [hunk 2]
```diff
@@ -137,6 +137,31 @@ def read(self, decode='utf-8', accept=[]):
...
 
+    def move(self, target_directory, new_name=None, overwrite=False):
+        """Move this file to the specified directory. The following parameters may be specified:
+
+        target_directory : string
+            Target directory where the file is to be moved.
+        new_name : string, optional
+            New name to be assigned to file (including extension, if any)
+        overwrite : boolean, optional
+            Specify if any already existing file needs to be overwritten.
```
Comment (apdavison) [inline hbp_archive.py]: "needs to be" --> "should be"

Comment (appukuttan-shailesh) [reply hbp_archive.py]: Done.

File: hbp_archive.py (hunk 1 above)
Comment (appukuttan-shailesh) [reply hbp_archive.py]: Ah yes... copy-paste error.

File: hbp_archive.py (hunk 2 above)
Comment (apdavison) [reply hbp_archive.py]: I think there are a couple of instances of this...

Comment (appukuttan-shailesh) [reply hbp_archive.py]: Corrected now.

**Comment by apdavison** [comment]
Thanks for implementing this!

**Pull request merged**